### PR TITLE
ENH: Add SIMD operations for min and max value comparision

### PR DESCRIPTION
**Pull request by howjmay**
Add SIMD operations for comparing maximum and minimum values.
`npyv_[max|min]_u64` is not implemented in this PR, since AVX2 has some problems on implementing them.

### CODE REVIEW

**Comment by seiko2plus** [comment]
@HowJMay, Could you please add a testing case for integer min/max?

**Comment by HowJMay** [comment]
@seiko2plus Is there anything I should fix or add?

**Review by seiko2plus — CHANGES_REQUESTED** [review]
The current testing case only tests f32/f64, you have to add a new test case for integer max/min at the tail of the following class

https://github.com/numpy/numpy/blob/c753b23e84a5474c39999fec1f2a6d1ccd54b9a3/numpy/core/tests/test_simd.py#L158-L161

File: numpy/core/tests/test_simd.py [hunk 1]
```diff
@@ -208,14 +208,30 @@ def test_arithmetic_subadd_saturated(self):
...
+        data_b = self._data(self._int_min(), reverse=True)
+        vdata_a, vdata_b = self.load(data_a), self.load(data_b)
+
+        data_max = self._int_clip([max(a, b) for a, b in zip(data_a, data_b)])
+        simd_max = self.max(vdata_a, vdata_b)
+        assert simd_max == data_max
+
+        data_min = self._int_clip([min(a, b) for a, b in zip(data_a, data_b)])
+        simd_min = self.min(vdata_a, vdata_b)
+        assert simd_min == data_min
```
Comment (seiko2plus) [inline numpy/core/tests/test_simd.py:225]:
```suggestion
        data_a = self._data()
        data_b = self._data(self.nlanes)
        vdata_a, vdata_b = self.load(data_a), self.load(data_b)

        data_max = [max(a, b) for a, b in zip(data_a, data_b)]
        simd_max = self.max(vdata_a, vdata_b)
        assert simd_max == data_max

        data_min = [min(a, b) for a, b in zip(data_a, data_b)]
        simd_min = self.min(vdata_a, vdata_b)
        assert simd_min == data_min
```

**Review by seiko2plus — CHANGES_REQUESTED** [review]
fix unit test cases

**Review by seiko2plus — APPROVED** [review]
LGTM, Thank you.

**Comment by HowJMay** [comment]
> LGTM, Thank you.

Thank you so much for helping

**Comment by charris** [comment]
Thanks @HowJMay .

**Pull request merged**